libraries-2/src/test/java/com/baeldung/freebuilder/EmployeeBuilderUnitTest.java

```diff
@@ -0,0 +1,164 @@
+package com.baeldung.freebuilder;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import com.baeldung.freebuilder.builder.classic.Address;
+
+public class EmployeeBuilderUnitTest {
+
+    private static final int PIN_CODE = 223344;
+    public static final String CITY_NAME = "New York";
+    public static final int INPUT_SALARY_EUROS = 10000;
+    public static final double EUROS_TO_USD_RATIO = 0.6;
+
+    @Test
+    public void whenBuildEmployeeWithAddress_thenReturnEmployeeWithValidAddress() {
+
+        // when
+        Address.Builder addressBuilder = new Address.Builder();
+        Address address = addressBuilder.setCity(CITY_NAME).build();
+
+        Employee.Builder builder = new Employee.Builder();
+
+        Employee employee = builder.setName("baeldung").setAge(10).setDesignation("author").setEmail("abc@xyz.com").setSupervisorName("Admin").setPhoneNumber(4445566).setPermanent(true).setRole("developer").setAddress(address).build();
```

Comment:
Let's break these builder calls up, one method per line, like you did below starting on line 47.